packages/fuels/tests/test_projects/options/src/main.sw

```diff
@@ -0,0 +1,90 @@
+contract;
+
+use std::{
+    address::Address,
+    constants::ZERO_B256,
+    option::Option,
+};
+
+struct TestStruct {
+    option: Option<Address>
+}
+
+enum TestEnum {
+    EnumOption: Option<Address>
+}
+
+abi MyContract {
+    fn get_some_u64() -> Option<u64>;
+    fn get_some_address() -> Option<Address>;
+    fn get_some_struct() -> Option<TestStruct>;
+    fn get_some_enum() -> Option<TestEnum>;
+    fn get_some_tuple() -> Option<(TestStruct, TestEnum)>;
+    fn get_none() -> Option<Address>;
+    fn input_primitive(s: Option<u64>) -> bool;
+    fn input_struct(s: Option<TestStruct>) -> bool;
+    fn input_enum(e: Option<TestEnum>) -> bool;
+    fn input_none(none: Option<Address>) -> bool;
+}
+
+impl MyContract for Contract {
+    fn get_some_u64() -> Option<u64> {
+        Option::Some(10)
+    }
+
+    fn get_some_address() -> Option<Address> {
+        Option::Some(~Address::from(ZERO_B256))
+    }
+
+    fn get_some_struct() -> Option<TestStruct> {
+        Option::Some(TestStruct{option: Option::Some(~Address::from(ZERO_B256))})
+    }
+
+    fn get_some_enum() -> Option<TestEnum> {
+        Option::Some(TestEnum::EnumOption(Option::Some(~Address::from(ZERO_B256))))
+    }
+
+    fn get_some_tuple() -> Option<(TestStruct, TestEnum)> {
+        let s = TestStruct{option: Option::Some(~Address::from(ZERO_B256))};
+        let e = TestEnum::EnumOption(Option::Some(~Address::from(ZERO_B256)));
+        Option::Some((s,e))
+    }
+
+    fn get_none() -> Option<Address> {
+        Option::None
+    }
+
+    fn input_primitive(input: Option<u64>) -> bool{
+        if let Option::Some(u) = input {
+            return u == 36;
+        }
+        false
+    }
+
+    fn input_struct(input: Option<TestStruct>) -> bool {
+        if let Option::Some(s) = input {
+            if let Option::Some(a) = s.option {
+                return a == ~Address::from(ZERO_B256);
```

Comment:
I think this test should be with an address different from the 0 address, just as a sanity check.

Reply:
We are testing that in other parts of the code but I agree. For sanity let's change it to something :smile: